src/transformers/integrations/deepspeed.py

```diff
@@ -304,11 +306,18 @@ def deepspeed_optim_sched(trainer, hf_deepspeed_config, args, num_training_steps
         lr_scheduler = DummyScheduler(optimizer)
     else:
         if isinstance(optimizer, DummyOptim):
-            raise ValueError(
-                "Found `optimizer` configured in the DeepSpeed config, but no `scheduler`. "
-                "Please configure a scheduler in the DeepSpeed config."
-            )
-        lr_scheduler = trainer.create_scheduler(num_training_steps=num_training_steps, optimizer=optimizer)
+
+            def _lr_scheduler_callable(optimizer):
```

Comment:
Might not matter in this context, but by assigning this locally defined function, the `lr_scheduler` becomes unpickleable.